jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessor.java

```diff
@@ -49,4 +67,97 @@ public static JarType determineJarType(Path jarPath) throws IOException {
     }
     return JarType.STANDARD;
   }
+
+  /**
+   * Explode jar and create three layers for classes, resources and dependencies on container.
+   *
+   * @param jarPath path to jar file.
+   * @param tempDirPath path to temporary jib local directory to use.
+   * @return list of {@link FileEntriesLayer}.
+   * @throws IOException if I/O error occurs when opening the jar file or if the directory for the
+   *     temporary directory path provided, doesn't exist.
+   */
+  public static List<FileEntriesLayer> explodeStandardJar(Path jarPath, @Nullable Path tempDirPath)
+      throws IOException {
+    Path localExplodedJarRoot;
+    if (tempDirPath == null) {
+      try (TempDirectoryProvider tempDirectoryProvider = new TempDirectoryProvider()) {
+        localExplodedJarRoot = tempDirectoryProvider.newDirectory();
+      }
+    } else {
+      localExplodedJarRoot = tempDirPath;
+    }
+    ZipUtil.unzip(jarPath, localExplodedJarRoot);
+
+    List<FileEntriesLayer> layers = new ArrayList<>();
+    Predicate<Path> isClassFile = path -> path.getFileName().toString().endsWith(".class");
+    Predicate<Path> isResourceFile = isClassFile.negate();
+
+    // Determine class and resource files in the directory containing jar contents and create
+    // FileEntriesLayer for each type of layer (class or resource), while maintaining the
+    // file's original project structure.
+    FileEntriesLayer classesLayer =
+        addDirectoryContentsToLayer(
+                FileEntriesLayer.builder(),
+                localExplodedJarRoot,
+                isClassFile,
+                APP_ROOT.resolve(RelativeUnixPath.get("explodedJar")))
+            .setName("Classes")
+            .build();
+    FileEntriesLayer resourcesLayer =
+        addDirectoryContentsToLayer(
+                FileEntriesLayer.builder(),
+                localExplodedJarRoot,
+                isResourceFile,
+                APP_ROOT.resolve(RelativeUnixPath.get("explodedJar")))
+            .setName("Resources")
+            .build();
+
+    // Get dependencies from Class-Path in the jar's manifest and create a FileEntriesLayer.Builder
+    // with these dependencies as entries.
+    List<Path> dependencies;
+    JarFile jarFile = new JarFile(jarPath.toFile());
+    String classPath =
+        jarFile.getManifest().getMainAttributes().getValue(Attributes.Name.CLASS_PATH);
+    if (classPath != null) {
+      dependencies =
+          Splitter.onPattern("\\s+")
+              .splitToList(classPath.trim())
+              .stream()
+              .map(Paths::get)
+              .collect(Collectors.toList());
+    } else {
+      dependencies = new ArrayList<>();
```

Comment:
You can remove this `else` branch if you initialize the variable on line 118 like `List<Path> dependencies = new ArrayList<>();`.

How about skipping adding the dependencies layer if it's empty? To be fair, it shouldn't matter, because I believe jib-core will drop any empty layers. But we can be conservative at this level too.

Reply:
That's good to know. Modified this to not add a dependency layer if there are no dependencies specified in `Class-Path`.